### PR TITLE
Basic info

### PR DESCRIPTION
**Pull request by egonelbre**

### CODE REVIEW

File: code/Style.md [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+
+Use properly namespaced `*zap.Logger`. By properly namespacing and passing in the logger we can better find how things are working. Tests should use `zaptest.NewLogger(t)` as the root logger.
+
+Avoid using global loggers (e.g. `zap.L().Error`, `log.Print`, `fmt.Print`), unless it is `package main`.
+
+## Variable naming
+
+Avoid single letter receivers. Prefer hard-to-confuse variables with 3-7 letters. Use conventional naming when there is one (e.g. `sync.Mutex` is usually called `mu`.) For variables a single letter variable can be fine only if the scope doesn't exceed one page (about 50 lines).
+
+Single letter receivers can be convenient, however when moving between a lot of packages it can be quickly confusing.
```
Comment (jtolds) [inline code/Style.md]: this isn't idiomatic advice: https://talks.golang.org/2014/names.slide#11. i'm okay with ignoring idiomatic advice but then i think i have a stronger case for 80 char column width limits :)

Comment (jtolds) [reply code/Style.md]: aslo https://research.swtch.com/names

File: code/Style.md [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+```
+
+## Type and method naming
+
+Consider package name as part of the type name, this avoids stutter when using the types or methods.
+
+```
+identity.FullIdentity // not great due to stutter
+identity.Full         // better
+```
```
Comment (jtolds) [inline code/Style.md]: hear hear. haven't fixed this yet

File: code/Style.md (hunk 1 above)
Comment (egonelbre) [reply code/Style.md]: Yeah, I'm aware, however given the current project I think erring on the side of slightly longer names doesn't cause problems.

**Pull request merged**